### DISCUSSION TITLE
Exclude Trait Parts (Hide Back Hair) When Other Traits Is Available (Helmet)

### DISCUSSION THREAD

**Discussion by rairulyle** [nftchef/art-engine#243] [Help + Q&A]
Hi!

I am using z-index method and I don't know how will I be able to exclude the back hair (Hair has 3 parts: Back, Main, Front) when a helmet trait was generated.

See below issue:
![image](https://user-images.githubusercontent.com/12708526/190022856-1c1b82d7-7aa7-43f8-acf1-fe0abe219781.png)

**Folder/File Structure:**
**Hair:**
![image](https://user-images.githubusercontent.com/12708526/190022903-c3df7153-f00b-44b0-8acf-39aeabd58f40.png)
**Helmet/Headgear:**
![image](https://user-images.githubusercontent.com/12708526/190022930-b9c3f6d1-77ec-493a-aeb6-e864b2e8857e.png)

**Comment by nftchef** [comment]
This is a use case for either adjusting your folder structure to "branch" into _either_ the helmet directory OR the _hair_ directory.
As in the following example, 
```
    layersOrder: [
      {
        name: "entry",
        sublayerOptions: {
          "z-1,Platform": { opacity: 0.5, bypassDNA: true },

          hoodTops: { trait: "Top" },
        },
      },
    ],
 ```
<img width="984" alt="image" src="https://user-images.githubusercontent.com/91582112/190860106-b94e2176-e596-431c-9f61-f022bd70acfa.png">

Here, I use only a single entry point in the layer config and use folders to select what should stack. In the `hoodTops#180` folder here would be the same as your Helmets and `regular2192` would be everything else, like hair

### Using incompatible 
You could choose to use the incompatible setting instead; although it may be harder to configure. It may take some testing and "advanced" configuration to set it up to skip the hair directory if a helmet is chosen or skip the helmet directory if the hair is chosen, depending on your layer config.

**Reply by rairulyle** [reply]
@nftchef , I mean my hair does have 3 parts, I want to choose the hair set still but I just want to exclude the back hair. Is it possible?

And I should mention that they're separate trait so my metadata looks like this. If I do what you said, I think I'll only get one trait. See json below.

``` json
 "attributes": [
    {
      "trait_type": "Faction",
      "value": "Celestial"
    },
    {
      "trait_type": "Background",
      "value": "Golden"
    },
    {
      "trait_type": "Hair",
      "value": "Long Brown"
    },
    {
      "trait_type": "Body Type",
      "value": "Pink"
    },
    {
      "trait_type": "Costume",
      "value": "Valkyrie"
    },
    {
      "trait_type": "Mouth",
      "value": "Model"
    },
    {
      "trait_type": "Eyes",
      "value": "Fiery Sun"
    },
    {
      "trait_type": "Headgear",
      "value": "Golden Halo"
    }
  ],
```
